### PR TITLE
Implements dry run feature

### PR DESCRIPTION
**Pull request by ramesius**

### CODE REVIEW

**Comment by ramesius** [comment]
This branch was made from the `refactor` branch, so #11 will need to be sorted out first.

**Comment by ramesius** [comment]
I will also try and get some more tests into the test suite for this new feature

File: src/start.js [hunk 1]
```diff
@@ -5,53 +5,67 @@ const startInstances = require('./instances/startInstances');
...
       startableInstances.forEach((instance) => {
         console.log(instance);
       });
-      return startInstances(startableInstances);
-    })
-    .then((startedInstances) => {
-      console.log('Finished starting instances. Moving on to untag them.');
-      return untagInstances(startedInstances);
+
+      return startInstances(startableInstances).then((startedInstanceIds) => {
```
Comment (krutisfood) [inline src/start.js]: If instances were previously started & the app stopped for what-ever reason this removes the ability to have Hammertime remove the tags.

File: src/stop.js [hunk 2]
```diff
@@ -5,57 +5,74 @@ const listInstancesToStop = require('./instances/listInstancesToStop');
 const tagInstances = require('./instances/tagInstances');
 const stopInstances = require('./instances/stopInstances');
 
-function spinDownASGs() {
+function stopAllInstances(dryRun) {
```
Comment (krutisfood) [inline src/stop.js]: Looks like you've moved this function up, was this intentional?

File: src/start.js [hunk 3]
```diff
@@ -5,53 +5,67 @@ const startInstances = require('./instances/startInstances');
...
       startableASGs.forEach((asg) => {
         console.log(asg.AutoScalingGroupName);
       });
-      return startASGs(startableASGs);
-    })
-    .then((startedASGs) => {
-      console.log(`Finished spinning up ASGs. Moving on to untag ${startedASGs.length} of them.`);
-      return untagASGs(startedASGs);
+
+      return startASGs(startableASGs).then((startedASGs) => {
```
Comment (krutisfood) [inline src/start.js]: Same comment as with instances.

File: src/stop.js (hunk 2 above)
Comment (ramesius) [inline src/stop.js]: Just cause

File: serverless.yml [hunk 4]
```diff
@@ -26,6 +26,10 @@ provider:
         - "logs:CreateLogStream"
         - "logs:PutLogEvents"
       Resource: "arn:aws:logs:*:*:*"
+    environment:
```
Comment (krutisfood) [inline serverless.yml]: Got too many spaces here, this should be a level back indenting wise (along with the line below too)

File: serverless.yml [hunk 5]
```diff
@@ -34,12 +38,12 @@ functions:
     events:
       - schedule:
           rate: cron(30 9 * * ? *) # ADST 8:30PM, AEST 7:30PM every day
-          enabled: ${file(./schedule.js):enabled}
+          enabled: ${self:custom:enabled}
```
Comment (krutisfood) [inline serverless.yml]: This leads to 
```
 Serverless Warning --------------------------------------
 
     A valid service attribute to satisfy the declaration
     'self:custom:enabled' could not be found.
 
 Serverless Warning --------------------------------------
 
     A valid service attribute to satisfy the declaration
     'self:custom:enabled' could not be found.
 ```

**Comment by krutisfood** [comment]
As discussed this doesn't change functionality, the filter works different to previous understanding.
Works well with dryRun true, just gotta test with it as false & figure out what's wrong with my git access :(

**Comment by krutisfood** [comment]
Magic, hammerTimeDryRun tested successfully in both `true` and `false` guises for stop, just awaiting the results of start in the morning.

**Pull request merged**